injector/ProgramProcessor.py

```
@@ -70,44 +70,36 @@ def run(self):
-                f.write(ast.unparse(currAst))
-
-        if SAVE_HEADER:
-            with open(os.path.join(self.outputDirectory, "header.json"), "w+") as f:
-                f.write(json.dumps(header))
\ No newline at end of file
+        try:
+            header_path = os.path.join(self.outputDirectory, "header.json")
+            with open(header_path, "w+") as f:
+                f.write(json.dumps(header, indent=2))
+        except IOError as e:
+            raise RuntimeError(f"Failed to write header.json: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)